accord-core/src/main/java/accord/messages/Request.java

```diff
@@ -23,7 +23,16 @@
 
 public interface Request extends Message
 {
-    void process(Node on, Id from, ReplyContext replyContext);
     default long waitForEpoch() { return 0; }
     default long knownEpoch() { return waitForEpoch(); }
+    void preProcess(Node on, Id from, ReplyContext replyContext);
```

Comment:
This is used to facilitate an external error handling path that combines with `MapReduceConsume`.

Seems like it would be better interface wise to add a method that takes all these parameters and the error, but I can see how that would lead to duplication and wouldn't be a ton clearer.

Reply:
Yes. I've looked into alternatives and they seemed no better ¯\_(ツ)_/¯